File: Greedy/054.Jump-Game-II/054.Jump-Game-II.java
class Solution {
    public int jump(int[] nums) {
        if (nums.length == 1) {
            return 0;
        }
        int start = 0;
        int end = 0;
        int step = 0;
        while (start <= end) {
            int newEnd = end;
            for (int i = start; i <= end; i++) {
                newEnd = Math.max(newEnd, nums[i] + i);
                if (newEnd >= nums.length - 1) {
                    return step + 1;
                }
            }
            step += 1;
            start = end + 1;
            end = newEnd;
        }
        return -1;
    }
}

File: Greedy/054.Jump-Game-II/054.Jump-Game-II.py
class Solution:
    def jump(self, nums: List[int]) -> int:
        if len(nums) == 1:
            return 0
        step = 0
        start = 0
        end = 0
        while start <= end:
            newEnd = end
            for i in range(start, end + 1):
                newEnd = max(newEnd, nums[i] + i)
                if newEnd >= len(nums) - 1:
                    return step + 1
            step += 1
            start = end + 1
            end = newEnd
        return -1
